### ISSUE TITLE
Maven ITs fail out-of-the-box

### ISSUE THREAD

**Issue by rombert** [adobe/aemanalyser-maven-plugin#138]
### Expected Behaviour

Running `mvn clean install` should be successful.

### Actual Behaviour

Running `mvn clean install` fails due to the Maven plug-in ITs not running properly.

```
[INFO] --- maven-invoker-plugin:3.2.2:verify (integration-test) @ aemanalyser-maven-plugin ---
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 0, Failed: 2, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[ERROR] The following builds failed:
[ERROR] *  content-package-files-test/pom.xml
[ERROR] *  wknd-test/pom.xml
[INFO] -------------------------------------------------
```

The error that is reported is

```
[ERROR] Failed to execute goal com.adobe.aem:aemanalyser-maven-plugin:1.4.0-SNAPSHOT:project-analyse (aem-analyser) on project aemanalyser-maven-plugin-it-content-package-files-test: Failed to resolve version for com.adobe.aem:aemanalyser-maven-plugin:jar:RELEASE -> [Help 1]
```

This is with 
```
$ git describe HEAD
aemanalyser-maven-plugin-1.3.0-9-gde8887f
```

**Comment by rombert** [comment]
@bosschaert @cziegeler @kwin - you have recently contributed to the plug-in. Do the tests work for you?

**Comment by cziegeler** [comment]
yes, works for me without any problems

**Comment by bosschaert** [comment]
Works for me as well.

```
[INFO] AEM Parent Plugin Open Source 2-SNAPSHOT ........... SUCCESS [  0.437 s]
[INFO] AEM Analyser Core 1.1.3-SNAPSHOT ................... SUCCESS [  3.913 s]
[INFO] AEM Analyser Maven Plugin 1.4.0-SNAPSHOT ........... SUCCESS [01:36 min]
[INFO] AEM Analyser Maven Plugin Project Reactor 0.0.1-SNAPSHOT SUCCESS [  0.061 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

**Comment by kwin** [comment]
Please share a debug log, RELEASE version is fully supported in https://github.com/apache/maven/blob/6ca13af230e9a7c6dcb0bf629d9a2676d8f833ed/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultVersionResolver.java#L131. It is just that you shouldn't use it as regular Maven dependency version.

**Comment by kwin** [comment]
Looks like this is the issue here: 
> [DEBUG] Skipped remote request for com.adobe.aem:aemanalyser-maven-plugin/maven-metadata.xml locally installed metadata up-to-date

(triggered in https://github.com/apache/maven-resolver/blob/e5c8763dfe390a537911094810d0bcf4c58180eb/maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/DefaultUpdateCheckManager.java#L258). Probably the check policy needs some update here.

**Comment by rombert** [comment]
@kwin - I filed a PR that fixes the tests for me ( #140 ). Not sure it's the best approach though. I have maven-metadata entries from multiple repositories, but perhaps the one that trips the resolution is `maven-metadata-local.xml`

```xml
<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.adobe.aem</groupId>
  <artifactId>aemanalyser-maven-plugin</artifactId>
  <versioning>
    <latest>1.4.0-SNAPSHOT</latest>
    <versions>
      <version>1.1.9-SNAPSHOT</version>
      <version>1.1.11-SNAPSHOT</version>
      <version>1.4.0-SNAPSHOT</version>
    </versions>
    <lastUpdated>20220208171146</lastUpdated>
  </versioning>
</metadata>
```

Note that it does not have a `<release>` entry. All other local entries have it.

**Comment by cziegeler** [comment]
@rombert I'm wondering if a normal usage of the aemanalyser plugin fails for you as well? For example if you generate a new project with the latest aem archetype and run it, does it fail with the same error?
So the question is, do we just need to fix the ITs or is it a general problem?

**Comment by kwin** [comment]
I think to be on the safe side, we should adjust the update policy for this request. I will try to come up with a PR. This seems like a edge case which is not handled particularly well in the Maven Resolver. Will later on open a bug for that as well.

**Comment by rombert** [comment]
@cziegeler - if I generate a project using the [aem-project-archetype](https://github.com/adobe/aem-project-archetype) it works just fine for me. The build works with both the 1.3.0 release and the 1.4.0-SNAPSHOT development version.

@kwin - I'm in the end fine with whichever solution is chosen, as long as I can run the ITs :-) I'll comment on #141 regarding the implementation. Thanks!

**Comment by rombert** [comment]
To make things even more fun, now the ITs pass for me. Not sure why that has changed, perhaps the retry interval has passed.

**Comment by kwin** [comment]
Exactly, your last update of your local metadata is more than 24 hours old so the update policy "daily" kicks in!
